### DISCUSSION TITLE
Fleet 404 error on new distributed install

### DISCUSSION THREAD

**Discussion by dgaudettesn** [Security-Onion-Solutions/securityonion#6512] [Unsupported Versions] [answered]
This is a new on-prem installation of 2.3.90 from the Security Onion ISO.  It's 3 nodes -- 1 manager, 1 search, 1 forward, and all settings for each node during setup were default except the docker IP address range was changed.

So-status will sometimes show all services running fine, but sometimes so-fleet will be in error.  Grid will also intermittently switch from OK to Fault on the Manager.

When I click on the FleetDM link on the Manager web page, I always get a 404 error message.

Fleet output from running sudo salt-call state.highstate is below, and the fleet log directory is empty.  I do notice the subnet mask from the salt-call output does not match the expected /24 mask.  

I've tried some solutions mentioned in other posts but havne't had any luck.  I've re-installed Security Onion on all nodes, and soup shows everything is up to date.

----------
          ID: fleetcdir
    Function: file.directory
        Name: /opt/so/conf/fleet/etc
      Result: True
     Comment: The directory /opt/so/conf/fleet/etc is in the correct state
     Started: 14:14:06.072457
    Duration: 1.712 ms
     Changes:   
----------
          ID: fleetpackcdir
    Function: file.directory
        Name: /opt/so/conf/fleet/packs
      Result: True
     Comment: The directory /opt/so/conf/fleet/packs is in the correct state
     Started: 14:14:06.074298
    Duration: 1.464 ms
     Changes:   
----------
          ID: fleetnsmdir
    Function: file.directory
        Name: /nsm/osquery/fleet
      Result: True
     Comment: The directory /nsm/osquery/fleet is in the correct state
     Started: 14:14:06.075886
    Duration: 1.618 ms
     Changes:   
----------
          ID: fleetpacksync
    Function: file.recurse
        Name: /opt/so/conf/fleet/packs
      Result: True
     Comment: The directory /opt/so/conf/fleet/packs is in the correct state
     Started: 14:14:06.077621
    Duration: 139.684 ms
     Changes:   
----------
          ID: fleetpackagessync
    Function: file.recurse
        Name: /opt/so/conf/fleet/packages
      Result: True
     Comment: The directory /opt/so/conf/fleet/packages is in the correct state
     Started: 14:14:06.217465
    Duration: 729.108 ms
     Changes:   
----------
          ID: fleetlogdir
    Function: file.directory
        Name: /opt/so/log/fleet
      Result: True
     Comment: The directory /opt/so/log/fleet is in the correct state
     Started: 14:14:06.946722
    Duration: 1.578 ms
     Changes:   
----------
          ID: fleetdb
    Function: mysql_database.present
        Name: fleet
      Result: True
     Comment: Database fleet is already present
     Started: 14:14:06.950014
    Duration: 9.957 ms
     Changes:   
----------
          ID: fleetdbuser
    Function: mysql_user.present
      Result: True
     Comment: User fleetdbuser@192.168.3.0/255.255.0.0 is already present with the desired password
     Started: 14:14:06.968883
    Duration: 15.984 ms
     Changes:   
----------
          ID: fleetdbpriv
    Function: mysql_grants.present
      Result: True
     Comment: Grant all privileges on fleet.* to fleetdbuser@192.168.3.0/255.255.0.0 is already present
     Started: 14:14:06.987143
    Duration: 6.207 ms
     Changes:   
----------
          ID: so-fleet
    Function: docker_container.running
      Result: True
     Comment: Container 'so-fleet' is already configured as specified. State changed from 'stopped' to 'running'.
     Started: 14:14:06.999807
    Duration: 2594.768 ms
     Changes:   
              ----------
              state:
                  ----------
                  new:
                      running
                  old:
                      stopped
----------

          ID: append_so-fleet_so-status.conf
    Function: file.append
        Name: /opt/so/conf/so-status/so-status.conf
      Result: True
     Comment: File /opt/so/conf/so-status/so-status.conf is in correct state
     Started: 14:14:09.595330
    Duration: 7.795 ms
     Changes:   
----------

**Comment by defensivedepth** [comment]
Can you please post the output to the following command, run on your Manager (sanitize as needed) : 

`sudo salt-call pillar.get global`

`cat /opt/so/saltstack/local/salt/fleet/packages/launcher.flags`

**Comment by dgaudettesn** [comment]
sudo salt-call pillar.get global

local:
    ----------
    airgap:
        False
    cortexkey:
        oE123456789123qh25zj
    cortexorgname:
        SecurityOnion
    cortexorguserkey:
        x0a12345678912347aZ2
    cortexplaysecret:
        2NWS12345789123AqtSG
    dockernet:
        192.168.3.0
    fleet_custom_hostname:
        None
    fleet_hostname:
        somaster
    fleet_ip:
        10.20.132.32
    fleet_manager:
        True
    fleet_node:
        False
    fleet_packages-timestamp:
        2021-12-09-14:09
    fleet_packages-version:
        4
    hivekey:
        6Q123456789123n4jxoGj
    hiveplaysecret:
        Vml12345789312FaWy
    hnmanager:
        10.0.0.0/8,192.168.0.0/16,172.16.0.0/12
    ids:
        Suricata
    imagerepo:
        security-onion-solutions
    managerip:
        10.20.132.32
    mdengine:
        ZEEK
    pipeline:
        redis
    sensoronikey:
        nYh31234567891231cHS
    soversion:
        2.3.90
    url_base:
        10.20.132.32
    wazuh:
        1

[thisadmin@somaster ~]$ cat /opt/so/saltstack/local/salt/fleet/packages/launcher.flags
cat: /opt/so/saltstack/local/salt/fleet/packages/launcher.flags: Permission denied

[thisadmin@somaster ~]$ sudo cat /opt/so/saltstack/local/salt/fleet/packages/launcher.flags
enroll_secret_path /etc/so-launcher/secret
hostname 10.20.132.32:8090
root_directory /var/so-launcher/securityonion
osqueryd_path /usr/local/so-launcher/bin/osqueryd
root_pem /etc/so-launcher/roots.pem
autoupdate
update_channel stable

**Comment by defensivedepth** [comment, marked as answer]
@dgaudettesn I have confirmed this is a bug - appears to be something related to when the default docker IP range is changed.

I have created an issue to track resolution - https://github.com/Security-Onion-Solutions/securityonion/issues/6603

**Reply by defensivedepth** [reply]
This has now been fixed - https://github.com/Security-Onion-Solutions/securityonion/pull/6605/commits/a23824e19968d737d30f36767fc582410ff512fd

You can manually fix it yourself right now by editing the file as seen in the commit and then running the following command on the Manager, which will setup Fleet:

Email will be used as the username.

`sudo so-fleet-setup <your-email> <your-password>`
